datadog_lambda/tags.py

```diff
@@ -35,16 +35,27 @@ def parse_lambda_tags_from_arn(arn):
     split_arn = arn.split(":")
 
     # If ARN includes version / alias at the end, drop it
-    if len(split_arn) > 7:
-        split_arn = split_arn[:7]
+    if len(split_arn) <= 7:
 
-    _, _, _, region, account_id, _, function_name = split_arn
+        _, _, _, region, account_id, _, function_name = split_arn
 
-    return [
-        "region:{}".format(region),
-        "account_id:{}".format(account_id),
-        "functionname:{}".format(function_name),
-    ]
+        return [
+            "region:{}".format(region),
+            "account_id:{}".format(account_id),
+            "functionname:{}".format(function_name),
+        ]
+
+    else:
+        _, _, _, region, account_id, _, function_name, alias = split_arn
+
+        resource = function_name + "/" + alias
```

Comment:
I think this should be `function_name + ":" + alias`, based on how cloudwatch metrics with an alias are added. This is based on the metric aws.lambda.invocations with the tag resource:swf-hello-test:aws-alias on the. demo org. Also should resource be added in both cases, just in the first case without the alias. That would make it consistent with cloudwatch

Reply:
It's documented here https://docs.aws.amazon.com/lambda/latest/dg/monitoring-metrics.html. `resource` should be `functionname` (when function is not invoked by version or alias), or `functionname:version` (when invoked by version) or `functionname:alias` (when invoked by alias). The `executedversion` should only be present when the function is invoked by alias (i.e., when resource is `functionname:alias`), so it tells users which version is behind that alias.

That said, I believe ARN won't give you everything you need, you probably also need the `function_version` field from the `context` object https://docs.aws.amazon.com/lambda/latest/dg/python-context.html.